ocean_lib/ocean/util.py

```diff
@@ -71,22 +79,21 @@ def get_web3_connection_provider(network_url):
     :param network_url: str
     :return: provider : HTTPProvider
     """
-    if network_url == "ganache":
-        network_url = GANACHE_URL
-
     if network_url.startswith("http"):
         provider = CustomHTTPProvider(network_url)
-
+    elif network_url.startswith("ws"):
+        provider = WebsocketProvider(network_url)
+    elif network_url == "ganache":
```

Comment:
are we sure we need these 2 extra options? it seems to be the same condition as the "http" one. We could combine them.

Reply:
It's not quite the same. The "http" `if` handles user input like "http://...".These two options handle user input like "ganache" or "polygon". ganache and polygon are treated differently from the other names like "ropsten" because they're not supported by infura (at least they weren't at the time I did this code)